jscomp/core/lam_analysis.ml

```diff
@@ -27,6 +27,7 @@ let not_zero_constant (x : Lam_constant.t) =
   match x with
   | Const_int { i } -> i <> 0l
   | Const_int64 i -> i <> 0L
+  | Const_bigint i -> i <> "0"
```

Comment:
is this normalised, or are `-0` and `00` also possible?

Reply:
`00` is normalized, but `-0`. I'll add another clause `i <> "-0"`

Reply:
How about instead using a representation with a boolean (sign), and normalised string. So there's no doubt of the meaning and less likely to introduce bugs in future.

Reply:
Ah, very good. I'll refactor accordingly.